c2corg_ui/static/js/documentediting.js

```diff
@@ -374,6 +390,25 @@ app.DocumentEditingController.prototype.handleMapFeatureChange_ = function(
 
 
 /**
+ * It doesn't associate, it just pushes it into associations array.
+ * @private
+ */
+app.DocumentEditingController.prototype.pushDocToAssociations_ = function() {
+  var doctype = Object.keys(this.urlParams_)[0];
+  var url = '/search?q=' + this.urlParams_[doctype] + '&t=' + doctype[0];
+  // get the doc from the API like /search?q=12345&t=r  (routes)
+  this.api_.getJson_(url).then(function(doc) {
```

Comment:
searchById doesn't imply that you have to send and ID AND a doctype, but yes, we know that it will make the search quicker :) andSearchByIDandDoctype is hideous :D